src/restClient.js

```diff
@@ -8,23 +8,32 @@ import {
   DELETE,
 } from 'admin-on-rest/lib/rest/types';
 
-export default client => {
+export default (client, options = { id: 'id' }) => {
   const mapRequest = (type, resource, params) => {
     const service = client.service(resource);
     let query = {};
 
+    const mapParamsId = params => {
+      if(options.id !== 'id'){
+        params[options.id] = params.id;
+        delete params.id;
+      }
+      return params;
```

Comment:
@fonzarely  
You can start viewing this sample code here, it should work (please copy it and if you think its ok use it as base).

https://github.com/josx/aor-feathers-client/blob/fonzarely-custom-id/test/restClient.spec.js

https://github.com/josx/aor-feathers-client/commit/0559705b947dd75605ac82a2f2b897f1e23c4ace